libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp

```diff
@@ -422,21 +441,26 @@ void NavEKF3_core::SelectVelPosFusion()
     gpsDataToFuse = storedGPS.recall(gpsDataDelayed,imuDataDelayed.time_ms);
     if (gpsDataToFuse) {
         CorrectGPSForAntennaOffset(gpsDataDelayed);
+        // calculate innovations and variances for reporting purposes only
+        CalculateVelInnovationsAndVariances(gpsDataDelayed.vel, frontend->_gpsHorizVelNoise, frontend->gpsNEVelVarAccScale, gpsVelInnov, gpsVelVarInnov);
+    }
+
+    // detect position source changes
+    AP_NavEKF_Source::SourceXY pos_source = frontend->_sources.getPosXYSource();
+    if (pos_source != pos_source_last) {
+        pos_source_reset = (pos_source != AP_NavEKF_Source::SourceXY::NONE);
```

Comment:
Comment added.  It wouldn't be good to try and reset the position if there is no new position source.  This can happen if the user is switching between GPS and Optical flow.  Optical flow only provides velocity.